blockly-core/core/ui/block_space/block_space_editor.js

```diff
@@ -265,62 +265,55 @@ Blockly.BlockSpaceEditor.prototype.getDeleteAreas = function() {
  * 2. delete blocks on top of flyout
  * @private
  */
-Blockly.BlockSpaceEditor.prototype.bumpOrDeleteOutOfBoundsBlocks_ = function() {
+Blockly.BlockSpaceEditor.prototype.bumpBlocksIntoView_ = function() {
+  // Immediately return if dragging, to avoid expensive calculations
   if (Blockly.Block.isDragging()) {
     return;
   }
 
+  // Can't bump if we don't know our workspace dimensions
   var metrics = this.blockSpace.getMetrics();
-  if (!metrics || metrics.contentWidth > metrics.viewWidth ||
-      metrics.contentHeight > metrics.viewHeight) {
-    // Don't try to bump if there are no metrics, or if the viewWidth/Height is
-    // so small that the content overflows on both sides.
-    return;
-  }
-
-  var oneOrMoreBlocksOutOfBounds = metrics.contentTop < 0 ||
-    metrics.contentTop + metrics.contentHeight >
-    metrics.viewHeight + metrics.viewTop ||
-    metrics.contentLeft < (Blockly.RTL ? metrics.viewLeft : 0) ||
-    metrics.contentLeft + metrics.contentWidth >
-    metrics.viewWidth + (Blockly.RTL ? 2 : 1) * metrics.viewLeft;
-
-  if (!oneOrMoreBlocksOutOfBounds) {
+  if (!metrics) {
     return;
   }
```

Comment:
Since we only run this code when we drop a block anyway, it's probably not too bad to run through the blocks and do nothing without this global "content size vs view size" check - which is hard to handle gracefully in the "my block is wider than my workspace!" case anyway.   If folks are worried about perf, we can always put such a check back in.